src/clj/rems/workflow/permissions.clj

```diff
@@ -0,0 +1,166 @@
+(ns rems.workflow.permissions
+  (:require [clojure.set :as set]
+            [clojure.test :refer [deftest is testing]]))
+
+(def ^:private conj-set (fnil conj (hash-set)))
+
+(defn give-role-to-user [application role user]
+  (assert (keyword? role)
+          (str "role must be a keyword: " (pr-str role)))
+  (assert (string? user)
+          (str "user must be a string: " (pr-str user)))
+  (update-in application [::user-roles user] conj-set role))
+
+(defn give-role-to-users [application role users]
```

Comment:
Use only this plural version, no need for the singular function?

Reply:
Most of the usages give a role to a single user, `(:event/actor event)`. There are only two special cases (handlers and commenters) where there are multiple users. Why complicate the default case? There's also the operation in the other direction, `remove-role-from-user`, which doesn't even have a plural version. Should that also be made plural-only, even though it's not needed? And anyways for testing and code readability the singular versions would still exists, even if they would be private.

Reply:
I considered `give-role-to-users [application role & users]` but calling apply to pass in the list of users seemed too complicated.

Reply:
Simply because passing `[user]` is quite as simple as just `user`. And the resulting combined function is still only a few lines long.